doc/source/sphinxext/postprocess_api.py

```diff
@@ -0,0 +1,56 @@
+"""
+Sphinx plugin to run example scripts and create a gallery page.
+
+Lightly modified from the seaborn project (Michael Waskom).
+Originally, lightly modified from the mpld3 project.
+
+"""
+import os.path as op
+import glob
+from sphinx.application import Sphinx
+
+
+def on_build_finished(app: Sphinx, exception: Exception) -> None:
+    html_dir = op.abspath(app.builder.outdir)
```

Comment:
In general I'd advocate the usage of `pathlib.Path` and its OO-like API instead of the raw path manipulation functions from `os.path`. So, something like:

```
html_dir = Path(app.builder.outdir)
api_dir = html_dir / 'api'
[...]
index_html = html_dir / 'index.html'
with index_html.open("rt") as f:
   ...
[...]

for filename in sorted(api_dir.glob("*.html")):
    ...
```

It often turns out to be much more readable than the old-style path manipulation functions.

Reply:
yes yes you're right, just being lazy...